src/web/lib/useDocumentHidden.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+export function useDocumentVisibilityState(): VisibilityState {
+	const [visibilityState, setVisibilityState] = useState(
+		document.visibilityState,
+	);
+	const onVisibilityChange = () =>
+		setVisibilityState(document.visibilityState);
+
+	useEffect(() => {
+		document.addEventListener(
```

Comment:
I am really not sure if this is an issue or not but my PR radar is being pinged here by adding an event listener inside a useEffect. I guess one issue is it could be set and removed multiple times

Reply:
This is a fair point, and we discussed it with @MarSavar, and opted for using an empty array as the second param.

By adding logs we could actually confirm that it was fired up several times.